raidprotect/src/main.rs

```diff
@@ -55,4 +55,18 @@ mod translations {
     //! This module contains translations generated with [`rosetta-i18n`].
 
     rosetta_i18n::include_translations!();
+
+    /// Returns the desired language from the given argument.
+    /// Default language is english
+    /// Reffer to <https://discord.com/developers/docs/reference#locales> for locale names
+    /// Ex: 'fr' gives you the french language
+    impl From<&str> for Lang {
+        fn from(lang: &str) -> Self {
+            match lang {
```

Comment:
It would be better to match only on the language code (e.g. `en` in `en-GB`). This allows to match on any region variant (like `fr-FR` or `fr-CA`) without further modifications.

You can use [`split_once`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.split_once) to only match on characters before the `-`.